### DISCUSSION TITLE
Deepgram hangs infinitely

### DISCUSSION THREAD

**Discussion by khueGetvocal** [deepgram/community#789] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT Streaming

### Details

We have been using Deepgram STT in production for a few months. It was working really great until recently. The service has become quite unstable, hanging frequently in the middle of audio streaming (i.e., audio chunks are sent continuously but Deepgram stops sending the results back at some point). 

In some of the tasks we obtained the following kind of errors, for your information:

```bash
task: <Task pending name='Task-26855' coro=<LiveTranscription._receiver() running at /home/miniconda3/envs/py3/lib/python3.10/site-packages/deepgram/transcription.py:247> wait_for=<Future pending cb=[Task.task_wakeup()]>>
Task was destroyed but it is pending!
task: <Task pending name='Task-27749' coro=<LiveTranscription._receiver() running at /home/miniconda3/envs/py3/lib/python3.10/site-packages/deepgram/transcription.py:247> wait_for=<Future pending cb=[Task.task_wakeup()]>>
```

We use the following code to connect to Deepgram:

```python
async def connect_to_deepgram(transcript_received_handler) -> str:

    dict_dg_config = {
        "punctuate": True,
        "interim_results": True,
        "endpointing": 10,
        "encoding": "linear16",
        "sample_rate": 16000,
        "model": "nova-2",
        "smart_format":True,
        "vad_events":True,
        "utterance_end_ms":1000,
        "language": "fr",
        #"diarize":True,
    }

    try:
        socket = await DEEPGRAM_CLIENT.transcription.live(dict_dg_config)
        socket.registerHandler(
            socket.event.CLOSE, lambda c: print(f"Connection closed with code {c}.")
        )
        socket.registerHandler(
            socket.event.TRANSCRIPT_RECEIVED, transcript_received_handler
        )
        return socket
    except Exception as e:
        if os.environ["daemon_exec"] == "yes":
            service_log = ServiceLogs(
                id=generate_random_string(20),
                exec_URL=WEB_SERVER_DOMAIN,
                ts=round(time.time(), 2),
                traceback=str(e)[:500],
                service="transcribe_audio"
            )
            service_log.create_in_datastore()
        raise Exception(f"Could not open socket: {e}")
```

We have also tried using the V3 SDK official live transcription [example](https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/streaming/microphone/main.py) and observed the same instability. For example, sometimes it just hangs forever after printing the interim results:

```python
Interim Results: Asked me some
Interim Results: Ask me some question.
<nothing happens anymore>
```

We really hope that you could look into this issue. Unfortunately we had to temporarily switched to a competitor's service for obvious reasons, but we'd be happy to switched back once this issue has been resolved. 

Thank you very much in advance!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment, marked as answer]
@khueGetvocal 

 apologies for these issues you've been experiencing with Deepgram.

There are several backbone internet issues that we are aware of. To help root-cause your issue, could you please share the following:

*  What country or region are you making requests from? (There is a known network issue with India/southeast Asia)
* If you are sending URLs to audio files (rather than raw audio) to Deepgram, where is your audio stored? (There is a known but mitigated network issue with Google Cloud Storage)
* Can you please run this command and share the results: `traceroute api-alt.md1.deepgram.com`

**Reply by davidvonthenen** [reply]
The examples are known working references, so it's likely there is a networking issue

**Comment by davidvonthenen** [comment]
@khueGetvocal Can you run this example and see if it works for you?
https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/streaming/async_http/main.py
